packages/next/src/client/on-recoverable-error.ts

```diff
@@ -1,7 +1,7 @@
 import { NEXT_DYNAMIC_NO_SSR_CODE } from '../shared/lib/app-dynamic/no-ssr-error'
 
 export default function onRecoverableError(err: any, errorInfo: any) {
-  const digest = err.digest || errorInfo.digest
+  const digest = err.digest || (errorInfo && errorInfo.digest)
```

Comment:
```suggestion
  const digest = err.digest || errorInfo?.digest
```
Why don't you use optional chaining?